FlowCal/plot.py

```diff
@@ -1047,6 +1288,1578 @@ def hist1d(data_list,
+                    and (positive_edge[idx+1] != negative_edge[idx+1]):
+                # violin region is continuing
+                pass
+            elif (positive_edge[idx] != negative_edge[idx]) \
+                    and (positive_edge[idx+1] == negative_edge[idx+1]):
+                # violin region is closing
+                end = idx+1  # include this point
+                violin_regions.append(
+                    _ViolinRegion(positive_edge = positive_edge[start:end+1],
+                                  negative_edge = negative_edge[start:end+1],
+                                  height        = height[start:end+1]))
+                start = None  # we are no longer in a violin region
+            elif (positive_edge[idx] == negative_edge[idx]) \
+                    and (positive_edge[idx+1] == negative_edge[idx+1]):
+                # we are in an inter-region segment
+                start = None
+
+            idx += 1
+
+        if start is not None:
+            # if we were still in a violin region at the end,
+            # add the last region to the list
+            end = idx  # include this point
+            violin_regions.append(
+                _ViolinRegion(positive_edge = positive_edge[start:end+1],
+                              negative_edge = negative_edge[start:end+1],
+                              height        = height[start:end+1]))
+
+    # illustrate violin
+    if vert:
+        for vr in violin_regions:
+            plt.fill_betweenx(x1=vr.negative_edge,
+                              x2=vr.positive_edge,
+                              y=vr.height,
+                              **violin_kwargs)
+    else:
+        for vr in violin_regions:
+            plt.fill_between(y1=vr.positive_edge,
+                             y2=vr.negative_edge,
+                             x=vr.height,
+                             **violin_kwargs)
+
+    # illustrate summary statistic
+    if draw_summary_stat:
+        if scale == 'log':
+            positive_edge = np.log10(violin_position) + (violin_width/2.0)
+            negative_edge = np.log10(violin_position) - (violin_width/2.0)
+
+            positive_edge = 10**positive_edge
+            negative_edge = 10**negative_edge
+        else:
+            positive_edge = violin_position + (violin_width/2.0)
+            negative_edge = violin_position - (violin_width/2.0)
+
+        if vert:
+            plt.plot([negative_edge, positive_edge],
+                     [summary_stat, summary_stat],
+                     **draw_summary_stat_kwargs)
+        else:
+            plt.plot([summary_stat, summary_stat],
+                     [negative_edge, positive_edge],
+                     **draw_summary_stat_kwargs)
+
+def violin(data,
+           channel=None,
+           positions=None,
+           violin_width=None,
+           xscale=None,
+           yscale=None,
+           xlim=None,
+           ylim=None,
+           vert=True,
```

Comment:
There is no docstring for this new parameter.

Reply:
Ooo, good catch. Fixed in 0de9b02d086a5d9dd939a73af1050ca786a34a7d.